src/manager.rs

```diff
@@ -0,0 +1,212 @@
+use std::collections::HashMap;
+use std::path::{ Path, PathBuf };
+
+use ast::Script;
+use compile::ExecutableDevEnv;
+use parse::ParseError;
+use run::{ Execution, ExecutionEvent, Error as RunError, StartStopError };
+
+use foxbox_taxonomy::util::{ Id };
+use rusqlite;
+use transformable_channels::mpsc::{ channel, ExtSender, TransformableSender };
+
+/// A ScriptManager error.
+#[derive(Serialize, Debug)]
+pub enum Error {
+    /// The script you requested (by ID) does not exist.
+    NoSuchScriptError,
+    /// There was an error executing some SQL.
+    SQLError(String),
+    /// There was an error attempting to run a script. (See `run.rs`.)
+    RunError(RunError),
+    /// There was an error parsing the script's JSON.
+    ParseError(String),
+}
+
+/// A type for ensuring type-safety (Id<ScriptId>).
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Hash, Eq)]
+pub struct ScriptId;
+
+/// ScriptManager stores a persistent database of scripts and executes them.
+/// Each script can be individually enabled or disabled.
+/// When a script is enabled, it is always running (unless an error occured during launch).
+/// Script sources are stored as JSON strings in a SQLite database.
+pub struct ScriptManager<Env, T> where Env: ExecutableDevEnv + 'static {
+    env: Env,
+    /// The path to the SQLite file to store, e.g. "./database.sqlite"
+    path: PathBuf,
+    /// A map to track currently-executing scripts.
+    runners: HashMap<Id<ScriptId>, Execution<Env>>,
+    /// The tx end of the channel passed to ScriptManager::new()
+    tx: Box<T>
+}
+
+impl<Env, T> ScriptManager<Env, T> where Env: ExecutableDevEnv + 'static, T: ExtSender<(Id<ScriptId>, ExecutionEvent)> + TransformableSender<(Id<ScriptId>, ExecutionEvent)> {
+
+    /// Create a ScriptManager using a SQLite database file with the given path, i.e. filename.
+    /// If the database file does not exist, it will be created.
+    ///
+    /// NOTE: You MUST consume the contents of `tx` to prevent memory leaks.
+    ///
+    /// The database stores the raw script source, but only after the source has been parsed
+    /// to ensure validity.
+    pub fn new(env: Env, path: &Path, tx: Box<T>) -> Result<Self, Error> {
+
+        let connection = try!(rusqlite::Connection::open(&path));
+        try!(connection.execute("CREATE TABLE IF NOT EXISTS scripts (
+            id          TEXT NOT NULL PRIMARY KEY,
+            source      TEXT NOT NULL,
+            is_enabled  BOOL NOT NULL DEFAULT 1
+        )", &[]));
+
+        Ok(ScriptManager {
+            path: path.to_owned(),
+            env: env,
+            runners: HashMap::new(),
+            tx: tx
+        })
+    }
+
+    /// Load and launch all existing scripts from the database.
+    pub fn load(&mut self) -> Result<HashMap<Id<ScriptId>, Result<(), Error>>, Error> {
```

Comment:
Nit: You could use `ResultMap` to turn this into `Result<ResultMap<Id<ScriptId>, (), Error>, Error>`. This type is defined in `foxbox_taxonomy::api::ResultMap` (it might move later, but we'll refactor that when needed).